pyiceberg/table/snapshots.py

```diff
@@ -228,6 +229,15 @@ def __eq__(self, other: Any) -> bool:
         )
 
 
+@lru_cache
+def _manifests(io: FileIO, manifest_list: Optional[str]) -> List[ManifestFile]:
+    """Return the manifests from the manifest list."""
+    if manifest_list not in (None, ""):
+        file = io.new_input(manifest_list)  # type: ignore
+        return list(read_manifest_list(file))
+    return []
```

Comment:
I think this is the more pythonic way:
```suggestion
    if manifest_list:
```
Maybe better to do this check around line 260, and change the signature in:
```suggestion
def _manifests(io: FileIO, manifest_list: str) -> List[ManifestFile]:
    """Return the manifests from the manifest list."""
    file = io.new_input(manifest_list)  # type: ignore
    return list(read_manifest_list(file))
```